my-sms/app/models/course.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class Course < ActiveRecord::Base
+  attr_accessible :allocation, :description, :end_date, :name, :number_of_semesters, :start_date
+
+  validates :name, presence: true
+  validates :number_of_semesters, presence: true
+  validates :start_date, presence: true
+  validates :end_date, presence: true
+
+  validate :end_date_after_start
+
+  def end_date_after_start
+    return unless !end_date.blank? && !start_date.blank?
```

Comment:
What do you think of flipping the condition? I find the following form of this check more readable:

```ruby
return if end_date.blank? || start_date.blank?
```

As an alternative that I also think is a little more readable (but like a little less):

```ruby
return unless end_date.present? && start_date.present?
```

Reply:
I like that better too. 
Fixed in 5f3c7ddcae8e0aea443eb3eb0ab75e6ccb8868f0